src/main/java/com/sap/sgs/phosphor/fosstars/data/github/GitHubDataFetcher.java

```diff
@@ -148,7 +149,7 @@ public GitHubDataFetcher(GitHub github, Path base) throws IOException {
     this.base = base;
     this.github = github;
 
-    this.localRepositoriesInfo = loadLocalRepositoriesInfo();
+    this.localRepositoriesInfo = Collections.synchronizedMap(loadLocalRepositoriesInfo());
```

Comment:
Since you you `localRepositoriesInfo` is a synchronized block, is it necessary to use a synchronized map?

Reply:
I would say yes, otherwise it may lead to that in the `localRepositoryFor` method new items are added during the `cleanup` run.

Reply:
Then, should those methods also access the map in a synchronized block? I think that using both synchronized blocks and a synchronized map is redundand.

Reply:
Yes, with adding the additional `synchronized` blocks the wrapping is not mandatory.
But removing it also does not really change anything regarding performance (afaik).
And keeping it makes it at least clear / safe in case of modifications in the future.

Reply:
> Yes, with adding the additional `synchronized` blocks the wrapping is not mandatory.

Then, let's please avoid duplicating synchronization.

> But removing it also does not really change anything regarding performance (afaik).

Most likely we won't notice that, but it does.

> And keeping it makes it at least clear / safe in case of modifications in the future.

I am sorry but I don't see how it make it clearer and safer. Let's please avoid duplicating synchronization.

Reply:
> > But removing it also does not really change anything regarding performance (afaik).
> 
> Most likely we won't notice that, but it does.

Did not know this. Always thought performance is equal if the `synchronized` block is `synchronized` on the same object which is wrapped via the `Collections.synchronized...`.
And in our case the `localRepositoriesInfo`.
But thanks for this info 🙂

> > And keeping it makes it at least clear / safe in case of modifications in the future.
> 
> I am sorry but I don't see how it make it clearer and safer. Let's please avoid duplicating synchronization.

Done.